server/src/main/java/io/deephaven/server/runner/Main.java

```diff
@@ -88,29 +93,84 @@ public static Configuration init(String[] args, Class<?> mainClass) throws IOExc
+    private static Optional<SSLConfig> parseSSLConfig(String prefix, Configuration config, boolean needsIdentity) {
+        final Optional<Identity> identity = parseIdentityConfig(prefix, config);
+        if (needsIdentity && identity.isEmpty()) {
+            return Optional.empty();
+        }
+        final Builder builder = SSLConfig.builder();
+        identity.ifPresent(builder::identity);
+        parseTrustConfig(prefix, config).ifPresent(builder::trust);
+        parseClientAuth(prefix, config).ifPresent(builder::clientAuthentication);
+        return Optional.of(builder.build());
+    }
+
+    private static Optional<Identity> parseIdentityConfig(String prefix, Configuration config) {
+        final String identityType = config.getStringWithDefault(prefix(prefix, SSL_IDENTITY_TYPE), null);
+        if (identityType == null) {
             return Optional.empty();
         }
-        if (!PRIVATEKEY.equals(sslType)) {
+        if (!PRIVATEKEY.equals(identityType)) {
             throw new IllegalArgumentException(
                     String.format("Only support `%s` identity type through Configuration", PRIVATEKEY));
         }
-        String identityCa = config.getStringWithDefault(SSL_IDENTITY_CERT_CHAIN_PATH, null);
-        String identityKey = config.getStringWithDefault(SSL_IDENTITY_PRIVATE_KEY_PATH, null);
+        final String identityCa = config.getStringWithDefault(prefix(prefix, SSL_IDENTITY_CERT_CHAIN_PATH), null);
+        final String identityKey = config.getStringWithDefault(prefix(prefix, SSL_IDENTITY_PRIVATE_KEY_PATH), null);
         if (identityCa == null || identityKey == null) {
-            throw new IllegalArgumentException(String.format("Must specify `%s` and `%s`", SSL_IDENTITY_CERT_CHAIN_PATH,
-                    SSL_IDENTITY_PRIVATE_KEY_PATH));
+            throw new IllegalArgumentException(String.format("Must specify `%s` and `%s`",
+                    prefix(prefix, SSL_IDENTITY_CERT_CHAIN_PATH), prefix(prefix, SSL_IDENTITY_PRIVATE_KEY_PATH)));
         }
-        IdentityPrivateKey identity =
-                IdentityPrivateKey.builder().certChainPath(identityCa).privateKeyPath(identityKey).build();
-        return Optional.of(SSLConfig.builder().identity(identity).build());
+        return Optional.of(IdentityPrivateKey.builder().certChainPath(identityCa).privateKeyPath(identityKey).build());
+    }
+
+    private static Optional<Trust> parseTrustConfig(String prefix, Configuration config) {
+        final String trustType = config.getStringWithDefault(prefix(prefix, SSL_TRUST_TYPE), null);
+        if (trustType == null) {
+            return Optional.empty();
+        }
+        if (!CERTS.equals(trustType)) {
+            throw new IllegalArgumentException(
+                    String.format("Only support `%s` trust type through Configuration", CERTS));
+        }
+        final String trustPath = config.getStringWithDefault(prefix(prefix, SSL_TRUST_PATH), null);
+        if (trustPath == null) {
```

Comment:
opinionated, but why not `Optional.ofNullable(trustPath).map(TrustCertificates::of).orElseThrow(...)`? It seems silly to use optional and then do manual null checks?

Reply:
`orElseThrow` returns `T` - I *could* do:

`Optional.of(Optional.ofNullable(trustPath).map(TrustCertificates::of).orElseThrow(...))`, but that seems worse?

Reply:
or just or(() -> throw...)?

Reply:
Done.